### PR TITLE
Minor build improvements and fix Roslyn TSA upload

### PR DESCRIPTION
**Pull request by mregen**
- Roslyn SARIF files were not found by upload tool, add fix to common.props
- latest thirdpartynotices
- make getversion run without installing a .NET Core SDK (path is not set otherwise for -g option)
- skip codesignvalidation where obsolete

### CODE REVIEW

File: tools/scripts/get-version.ps1 [hunk 1]
```diff
@@ -9,9 +9,9 @@
 
 try {
     # Try install tool
-    & dotnet @("tool", "install", "-g", "nbgv") 2>&1 | Out-Null
+    & dotnet @("tool", "install", "--tool-path", "./tools", "--framework", "netcoreapp3.1", "nbgv") 2>&1 # | Out-Null
```
Comment (koepalex) [inline tools/scripts/get-version.ps1:12]: ># | Out-Null [](start = 105, length = 12)

Please delete dead code

File: common.props [hunk 2]
```diff
@@ -48,4 +48,7 @@
     <ReleaseVersion Condition="'$(PublicRelease)' == 'True'">$(VersionPrefix)</ReleaseVersion>
     <ReleaseVersion Condition="'$(PublicRelease)' == 'False'">$(NuGetPackageVersion)</ReleaseVersion>
   </PropertyGroup>
+  <PropertyGroup>
+    <ErrorLog>$(MSBuildProjectFullPath).$([System.Guid]::NewGuid().ToString().Substring(0,8)).sarif</ErrorLog>
```
Comment (koepalex) [inline common.props:52]: >.Substring(0,8) [](start = 77, length = 15)

Please check if we can live with longer filenames and avoid the substring

File: tools/templates/sdl.yml [hunk 3]
```diff
@@ -9,15 +9,12 @@ jobs:
     demands:
     - msbuild
     - npm
+  variables:
+    runCodesignValidationInjection: false
+    skipComponentGovernanceDetection: true
```
Comment (mregen) [inline tools/templates/sdl.yml:14]: >skipComponentGovernanceDetection [](start = 4, length = 32)

remove

**Review by koepalex — APPROVED** [review]
:shipit:

**Review by koepalex — APPROVED** [review]
:shipit:

**Pull request merged**